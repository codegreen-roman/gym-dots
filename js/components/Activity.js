import React from 'react'

export const Activity = () => {

    const now = new Date()

    return (
        <section style={{ border: '2px solid #9ACD32' }}>
            <div>rendered at {now.toLocaleTimeString()}</div>
        </section>
    )
}

