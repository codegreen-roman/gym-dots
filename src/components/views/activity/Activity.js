import React from 'react'
import { object } from 'prop-types'
import { Route, Switch, Redirect } from 'react-router-dom'
import { PreWorkout } from './preWorkout/PreWorkout'
import { Training } from './workout/Training'

export const Activity = ({ match }) => {
    return (
        <Switch>
            <Route path={`${match.url}/pre`} component={PreWorkout} />
            <Route path={`${match.url}/workout`} component={Training} />
            <Route path={`${match.url}/post`} component={
                () => <div>This is your complete stats</div>
            } />
            <Redirect from={`${match.url}/`} to={`${match.url}/pre`} />
        </Switch>
    )
}

Activity.propTypes = {
    match: object
}
