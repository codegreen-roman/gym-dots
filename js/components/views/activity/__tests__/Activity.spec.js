import React from 'react'
import { Activity } from '../Activity'
import { shallow } from 'enzyme'

const setup = () => {

    const props = {
        match: {
            url: '/activity'
        }
    }

    return {
        component: shallow(<Activity {...props} />),
        props
    }
}

describe('Activity component', () => {
    it('should hold 3 routes and 1 redirect', () => {
        const { component } = setup()
        expect(component.find('Route').length).toBe(3)
        expect(component.find('Redirect').length).toBe(1)
    })

    it('should match the snapshot', () => {
        const { component } = setup()
        expect(component).toMatchSnapshot()
    })
})
