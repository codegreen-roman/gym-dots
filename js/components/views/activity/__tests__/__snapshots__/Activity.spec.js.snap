// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreWorkout component should match the snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    path="/activity/pre"
  />
  <Route
    component={[Function]}
    path="/activity/workout"
  />
  <Route
    component={[Function]}
    path="/activity/post"
  />
  <Redirect
    from="/activity/"
    push={false}
    to="/activity/pre"
  />
</Switch>
`;
